src/test/controllerscriptenginelegacy_test.cpp

```diff
@@ -659,6 +659,52 @@ TEST_F(ControllerScriptEngineLegacyTest, connectionExecutesWithCorrectThisObject
     EXPECT_DOUBLE_EQ(1.0, pass->get());
 }
 
+TEST_F(ControllerScriptEngineLegacyTest, convertCharsetUndefinedOnUnknownCharset) {
+    const auto result = evaluate("engine.convertCharset('NULL', 'Hello!')");
+
+    EXPECT_EQ(qjsvalue_cast<QByteArray>(result), QByteArrayLiteral(""));
+}
+
+template<int N>
+QByteArray intByteArray(const char (&array)[N]) {
+    return QByteArray(array, N);
+}
+
+TEST_F(ControllerScriptEngineLegacyTest, convertCharsetCorrectValueWellKnown) {
+    const auto result = evaluate(
+            "engine.convertCharset(engine.WellKnownCharsets.Latin9, 'Hello!')");
+
+    // ISO-8859-15 ecoded 'Hello!'
+    EXPECT_EQ(qjsvalue_cast<QByteArray>(result),
+            intByteArray({0x48, 0x65, 0x6c, 0x6c, 0x6f, 0x21}));
+}
+
+TEST_F(ControllerScriptEngineLegacyTest, convertCharsetCorrectValueStringCharset) {
+    const auto result = evaluate("engine.convertCharset('ISO-8859-15', 'Hello!')");
+
+    // ISO-8859-15 ecoded 'Hello!'
+    EXPECT_EQ(qjsvalue_cast<QByteArray>(result),
+            intByteArray({0x48, 0x65, 0x6c, 0x6c, 0x6f, 0x21}));
+}
+
+TEST_F(ControllerScriptEngineLegacyTest, convertCharsetUnsupportedChars) {
+    auto result = qjsvalue_cast<QByteArray>(
+            evaluate("engine.convertCharset('ISO-8859-15', 'مايأ نامز')"));
+
+    // On some plateforms, conversion replaces unknown chars by '?' rather
+    // than the substitute character (https://en.wikipedia.org/wiki/Substitute_character)
+    // Both are valid results but we need a consistent one for the test.
+    for (int i = 0; i < result.length(); ++i) {
+        if (result[i] == 0x1A) {
+            result[i] = 0x3F;
+        }
+    }
```

Comment:
Or maybe I should just use the [`ConvertInvalidToNull`](https://doc.qt.io/qt-6/qstringconverter.html#Flag-enum) flag?

Reply:
If this is just relevant for <Qt6.4 than use an ifdef in the testcase as well.

Reply:
No need. `QTextCodecè [supports the `ConvertInvalidToNull` flag](https://doc.qt.io/qt-6/qtextcodec.html#makeEncoder).